packages/tpetra/core/src/Tpetra_Details_fill.hpp

```diff
@@ -83,6 +83,7 @@ class Fill {
   {
     static_assert (std::is_integral<IndexType>::value,
                    "IndexType must be a built-in integer type.");
+    // DEEP_COPY REVIEW - NOT TESTED
```

Comment:
As far as I can tell, this code is only used by MultiVector for implementing putScalar.

The only value this guy adds, I think is filling only certain columns of a MultiVector which is bigger.  I'm not sure KokkosBlas can do that (@brian-kelley Can it?).

This should all be scalar-to-device copies.


I'm sorely tempted to recommend rewriting this file (and/or dumping the functionality into MultiVector).